content/concepts/did-ddo.md

```diff
@@ -1,173 +1,602 @@
+    "algorithm": {
+      "language": "Node.js",
+      "version": "1.0.0",
+      "container": {
+        "entrypoint": "node $ALGO",
+        "image": "ubuntu",
+        "tag": "latest",
+        "checksum": "44e10daa6637893f4276bb8d7301eb35306ece50f61ca34dcab550"
+      }
+    }
+  }
+}
+```
 
-## DDO Service Types
+### Services
 
-There are many possible service types for a DDO.
+Services define the access for an asset, and each service is represented by its respective datatoken.
 
-- `metadata` - describing the asset
-- `access` - describing how the asset can be downloaded
-- `compute` - describing how the asset can be computed upon
+An asset should have at least one service to be actually accessible, and can have as many services which make sense for a specific use case.
 
-Each asset has a `metadata` service and at least one other service.
+| Attribute              | Type                        | Required                        | Description                                                                                                                                  |
+| ---------------------- | --------------------------- | ------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------- |
+| **`id`**               | `string`                    | **✓**                           | Unique ID                                                                                                                                    |
+| **`type`**             | `string`                    | **✓**                           | Type of service (`access`, `compute`, `wss`, etc.                                                                                            |
+| **`name`**             | `string`                    |                                 | Service friendly name                                                                                                                        |
+| **`description`**      | `string`                    |                                 | Service description                                                                                                                          |
+| **`datatokenAddress`** | `string`                    | **✓**                           | Datatoken address                                                                                                                            |
+| **`serviceEndpoint`**  | `string`                    | **✓**                           | Provider URL (schema + host)                                                                                                                 |
+| **`files`**            | [Files](#files)             | **✓**                           | Encrypted file URLs.                                                                                                                         |
```

Comment:
The spec says the `files` attribute is type `Files` which is described as an object:

```json
{
  "files": "0x044736da6dae39889ff570c34540f24e5e084f4e5bd81eff3691b729c2dd1465ae8292fc721e9d4b1f10f56ce12036c9d149a4dab454b0795bd3ef8b7722c6001e0becdad5caeb2005859642284ef6a546c7ed76f8b350480691f0f6c6dfdda6c1e4d50ee90e83ce3cb3ca0a1a5a2544e10daa6637893f4276bb8d7301eb35306ece50f61ca34dcab550b48181ec81673953d4eaa4b5f19a45c0e9db4cd9729696f16dd05e0edb460623c843a263291ebe757c1eb3435bb529cc19023e0f49db66ef781ca692655992ea2ca7351ac2882bf340c9d9cb523b0cbcd483731dc03f6251597856afa9a68a1e0da698cfc8e81824a69d92b108023666ee35de4a229ad7e1cfa9be9946db2d909735"
}
```

But then the "Full Enhanced DDO Example" at https://github.com/oceanprotocol/docs/pull/783/files#diff-bf71babd090b581340941e03f8ca2d7b5aa334359ecbf744c92bcaedf3e61de6R490  shows it as a string.

```json
"files": "0x044736da6dae39889ff570c34540f24e5e084f4e5bd81eff3691b729c2dd1465ae8292fc721e9d4b1f10f56ce12036c9d149a4dab454b0795bd3ef8b7722c6001e0becdad5caeb2005859642284ef6a546c7ed76f8b350480691f0f6c6dfdda6c1e4d50ee90e83ce3cb3ca0a1a5a2544e10daa6637893f4276bb8d7301eb35306ece50f61ca34dcab550b48181ec81673953d4eaa4b5f19a45c0e9db4cd9729696f16dd05e0edb460623c843a263291ebe757c1eb3435bb529cc19023e0f49db66ef781ca692655992ea2ca7351ac2882bf340c9d9cb523b0cbcd483731dc03f6251597856afa9a68a1e0da698cfc8e81824a69d92b108023666ee35de4a229ad7e1cfa9be9946db2d909735",
```

Which one is correct?  This should be fixed so that they match.

Reply:
fixed